UI/obs-app.cpp

```diff
@@ -2009,6 +2009,17 @@ static int run_program(fstream &logFile, int argc, char *argv[])
 	InstallNSApplicationSubclass();
 #endif
 
+#if defined(__linux__)
+	/* NOTE: Under GNOME Wayland session, Qt will always launch OBS with
+	 * X11 (xcb plarform). OBS need to force Qt by setting QT_QPA_PLATFORM
+	 * environment variable to "wayland". The user still can set
+	 * QT_QPA_PLATFORM or the -platform with an other option. */
+
+	const char *session_type = getenv("XDG_SESSION_TYPE");
+	if (session_type && strcmp(session_type, "wayland") == 0)
+		setenv("QT_QPA_PLATFORM", "wayland", false);
```

Comment:
I wonder if, on a super edgy edge case, if the user is on a GNOME / Wayland session but doesn't have the `qt5-wayland` package installed, this would lead to OBS not launching. It isn't necessarily a problem, since launching OBS on Xwayland won't work either, but is there something we can do here?

I've looked into it a bit, and there doesn't seem to have any easy way to list all Qt platforms, so I tend to declare this a non-problem for now.

Reply:
1. I thought about it when doing this https://github.com/obsproject/loganalyzer/pull/81.
If you try to launch it in a terminal Qt will write an error with a list:
```sh
$ obs -platform wyaland
qt.qpa.plugin: Could not find the Qt platform plugin "wyaland" in ""
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, wayland-egl, wayland, wayland-xcomposite-egl, wayland-xcomposite-glx, xcb.

[1]    8732 abort (core dumped)  obs -platform wyaland
```
Yes there is a typo because I already have qt5-wayland but the error will be the same.

2. I consider that packagers messed up by not putting it as dependency for OBS.

Reply:
@jp9000 @WizardCM  @Gol-D-Ace this commit makes OBS or Linux (or FreeBSD) require as a dependency the `qt5-wayland` packages the Qt6 variant in the future to avoid missing platform error from Qt.

I hope you have no issue with that.

Reply:
So after thinking about this a bit more, I think you're right. If a distribution built OBS Studio with Wayland support enabled, it is safe to assume that the package must depend on qt5-wayland. In this case, I think the only remaining to this branch is:

```diff
-  #if !defined(_WIN32) && !defined(__APPLE__) && defined(USE_XDG)
+  #if !defined(_WIN32) && !defined(__APPLE__) && defined(USE_XDG) && defined(ENABLE_WAYLAND)
```

Reply:
Change done